File: sdks/thunderpay-sdk-cs/Account.cs
/**
 * @file Account.cs
 * @author Krisna Pranav
 * @brief Account functionalities
 * @version 1.0
 * @date 2024-01-19
 *
 * @copyright Copyright (c) 2024 ThunderPayment Developers, Krisna Pranav
 *
 */

using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace ThunderPay
{
    public class Account : FundingInstrument
    {
        [JsonIgnore]
        public static string resource_href
        {
            get
            {
                return "/accounts";
            }
        }
    }
}

File: thunderpay-api/src/main/java/org/thunderpay/billing/account/api/Account.java
/**
 * @file Account.java
 * @author Krisna Pranav
 * @brief Account
 * @version 1.0
 * @date 2024-11-25
 *
 * @copyright Copyright (c) 2024 ThunderPayment Developers, Krisna Pranav
 *
 */

package org.thunderpay.billing.account.api;

import org.thunderpay.billing.entitlement.api.Blockable;
import org.thunderpay.billing.util.entity.Entity;

public interface Account extends AccountData, ImmutableAccountData, Blockable, Entity {
    public MutableAccountData toMutableAccountData();
    
    public Account mergeWithDelegate(final Account delegate);
}